### PR TITLE
CMake support: Install and export

### PR DESCRIPTION
**Pull request by tian-lt**
## Key changes
1. Add install and export phases into the cmake configuration.
1. Release pipeline is going to check out a new branch to perform the release actions.
1. Fix some CRLF-ed files.

## Sample runs:
- https://github.com/tian-lt/proxy/actions/runs/2555501456
- https://github.com/tian-lt/proxy/tree/release/0.7.0

### CODE REVIEW

File: .github/workflows/pipeline-release.yml [hunk 1]
```diff
@@ -9,38 +9,66 @@ on:
...
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: create release branch
+      run: |
+        git checkout -b release/${{ github.event.inputs.version }}
+
+    - name: assign cmake version
+      run: "((Get-Content -Path CMakeLists.txt -Raw) -Replace 'VERSION 0.1.0 # local build version','VERSION ${{ github.event.inputs.version }}') | Set-Content -Path CMakeLists.txt"
```
Comment (mingxwa) [inline .github/workflows/pipeline-release.yml:22]: Is there a more elegant way to do this? I guess `Set-Content` will also change the encoding of `CMakeLists.txt`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+        VERSION 0.1.0 # local build version
+        LANGUAGES CXX)
 add_library(msft_proxy INTERFACE)
 target_compile_features(msft_proxy INTERFACE cxx_std_20)
-target_include_directories(msft_proxy INTERFACE .)
+target_include_directories(msft_proxy INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+                                                $<INSTALL_INTERFACE:include>)
 
+# install and export the project. project name - proxy
+include(GNUInstallDirs)
```
Comment (mingxwa) [inline CMakeLists.txt:12]: Shall we do this on Windows as well?

File: .github/workflows/pipeline-release.yml (hunk 1 above)
Comment (tian-lt) [reply .github/workflows/pipeline-release.yml]: There is some tool in GitHub Marketplace to replace strings, but they are not GitHub 1P repos. Any good idea?
Adding `NoNewLine` switch can stop `Set-Content` from appending an extra line.

File: CMakeLists.txt (hunk 2 above)
Comment (tian-lt) [reply CMakeLists.txt]: Yes, I'd suggest take the naming conventions from GNU even the repo is going to install on Windows.
We don't specify any absolute path in the install & export phase, so all the directories and file will get combined with `CMAKE_INSTALL_PREFIX` which is given by user during installation.


Details about GNUInstallDirs:
> Define GNU standard installation directories
> Provides install directory variables as defined by the [GNU Coding Standards](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html).

**Review by mingxwa — APPROVED** [review]
Nice work!

**Pull request merged**